### PR TITLE
bugfix: add support for 307 and 308 HTTP redirection

### PR DESCRIPTION
**Pull request by jorgeaguileraseqera**
Signed-off-by: Jorge Aguilera <jorge.aguilera@seqera.io>

### CODE REVIEW

**Comment by pditommaso** [comment]
Minor thing the Path -> URI -> Path conversion conversion can be a bit fragile. what about 


```
    protected URLConnection toConnection(Path path) {
        final url = path.toUri().toURL()
        log.trace "File remote URL: $url"
        toConnection0(url, 0)   
    } 

    private URLConnection toConnection0(Path path, int attempt) {
          final conn = url.openConnection()
        conn.setRequestProperty("User-Agent", 'Nextflow/httpfs')
        if( url.userInfo ) {
            conn.setRequestProperty("Authorization", auth(url.userInfo));
        }
        else {
            XAuthRegistry.instance.authorize(conn)
        }
        if ( conn instanceof HttpURLConnection && conn.getResponseCode() in [307, 308]) {
            def header = conn.getHeaderFields()
            String location = readLocationHeader(header)
            return toConnection(new URL(location), attempt+1)
        }
        return conn
    }
```

**Comment by jorgeaguileraseqera** [comment]
perfect, also I've added a (missing) trace when redirect

File: modules/nf-httpfs/src/main/nextflow/file/http/XFileSystemProvider.groovy [hunk 1]
```diff
@@ -423,13 +436,18 @@ abstract class XFileSystemProvider extends FileSystemProvider {
...
+        if ( conn instanceof HttpURLConnection && conn.getResponseCode() in [200, 301, 302, 307, 308]) {
             def header = conn.getHeaderFields()
             return readHttpAttributes(header)
         }
         return null
     }
 
+    protected String readLocationHeader(Map<String,List<String>> header){
+        def location = header.get("Location")?.get(0)
+        location
```
Comment (pditommaso) [inline modules/nf-httpfs/src/main/nextflow/file/http/XFileSystemProvider.groovy:448]: Just `return header.get("Location")?.get(0)` ?

**Review by pditommaso — APPROVED** [review]
Looks great

**Pull request merged**